Classes/common/CDTReplicator/CDTReplicator.h

```diff
@@ -178,10 +193,21 @@ typedef NS_ENUM(NSInteger, CDTReplicatorState) {
  * return NO. Only when in `CDTReplicatorStatePending` will replication.
  *
  * @param error the error describing a failure.
+ * @param taskGroup an optional dispatch_group_t allowing clients to wait for the completion
+ *                  of all replication tasks in the taskGroup before proceeding.
  * @return YES or NO depending on success.
  *
  * @see CDTReplicatorState
  */
+- (BOOL)startWithError:(NSError *__autoreleasing *)error taskGroup:(nullable dispatch_group_t)taskGroup;
```

Comment:
To follow traditional Objc method name I think we need to rename this to `-startWithTaskGroup:Error`